### PR TITLE
Infer shape in Dirichlet constructor

### PR DESCRIPTION
**Pull request by MCGallaspy**
See #824.

### CODE REVIEW

File: pymc3/distributions/multivariate.py [hunk 1]
```diff
@@ -80,9 +80,11 @@ class Dirichlet(Continuous):
         as a parent of Multinomial and Categorical nevertheless.
     """
     def __init__(self, a, transform=transforms.stick_breaking, *args, **kwargs):
+        self.k = shape = a.shape[0]
+        if shape not in kwargs.keys():
```
Comment (twiecki) [inline pymc3/distributions/multivariate.py]: Shouldn't that be `'shape'`?

Comment (MCGallaspy) [inline pymc3/distributions/multivariate.py]: Indeed so.

**Comment by twiecki** [comment]
Thanks, LGTM. Will merge once we figured out the travis issue.

**Comment by twiecki** [comment]
Can you rebase against master?

**Comment by MCGallaspy** [comment]
Done.

**Pull request merged**